exploration/0006-number-selection.md

```diff
@@ -0,0 +1,127 @@
+
+<details>
+	<summary>Metadata</summary>
+	<dl>
+		<dt>Contributors</dt>
+		<dd>@eemeli</dd>
+		<dt>First proposed</dt>
+		<dd>2023-09-06</dd>
+		<dt>Pull Request</dt>
+		<dd>#000</dd>
+	</dl>
+</details>
+
+## Objective
+
+Define how selection on numbers happens.
+
+## Background
+
+As discussed on numerous past occasions and explicitly identified in
+<a href="https://github.com/unicode-org/message-format-wg/pull/457">#457</a>,
+it would be good for both plural and non-plural match selection to be supported in MF2.
+
+MF1 supported selection on either cardinal plurals or ordinal numbers,
+via the `plural` and `selectordinal` selectors.
+It also customized this selection beyond the capabilities of `com.ibm.icu.text.PluralRules`
+by allowing for an `offset` parameter.
+
+As pointed out by <a href="https://github.com/mihnita">@mihnita</a> in particular,
+category selection is not always appropriate for selection on a number:
+the number may be representing some completely other quantity,
+such as a four-digit year or the integer value of an enumerator.
+
+Furthermore, as pointed out by <a href="https://github.com/ryzokuken">@ryzokuken</a>
+in <a href="https://github.com/unicode-org/message-format-wg/pull/457#discussion_r1307443288">#457 (comment)</a>,
+ordinal values are not truly "plural",
+even if they use categories with similar names and are determined by similar rules.
+
+## Use-Cases
+
+Put together, we have thus identified three different types of selection
+we would like to support on numeric values:
+
+- cardinal plural selection
+- ordinal selection
+- exact match selection
+
+## Requirements
+
+- Enable cardinal plural selection.
+- Enable ordinal number selection.
+- Enable exact match selection.
+- Support relevant formatting options, such as `minimumFractionDigits`.
+- Encourage developers to use the same options for formatting and selection.
+
+## Constraints
+
+ICU MF1 messages using `plural` and `selectordinal` should be representable in MF2.
+
+## Proposed Design
+
+Given that we already have a `:number`,
+it makes sense to add a `<matchSignature>` to it with an option
+
+```xml
+<option name="type" values="plural ordinal exact" default="plural" />
+```
+
+The default `plural` value is presuming that to be the most common use case,
+and it affords the least bad fallback when used incorrectly:
+Using "plural" for "exact" still selects exactly matching cases,
+whereas using "exact" for "plural" will not select LDML category matches.
+This might not be noticeable in the source language,
+but may cause problems in target languages that the original developer is not considering.
+
+Additional options such as `minimumFractionDigits` and others already supported by `:number`
+should also be supported.
+
+## Alternatives Considered
+
+An alternative approach to this problem could be to leave the `:number` `<matchSignature>` undefined,
+and to define three further functions, each with a `<matchSignature>`:
+
+- `:plural`
+- `:ordinal`
```

Comment:
Note that `ordinal` is both a selector and a formatter, e.g. 

| Value | Selector | Output |
|---|---|---|
| 1 | one | 1st |
| 2 | two | 2nd |
| 3 | few | 3rd |
| 4 | other | 4th |
|97 | other | 97th |
| 101 | one | 101st |

Reply:
If an implementation does support `:ordinal` as a formatter, then it should probably also support it as a selector. But I'm not convinced that we'll want to include that in the required set.

Reply:
Ordinal would be needed for MF1 compatibility, if by MF1 compatibility we mean "ICU". But the callout is, I think, important, as you only mentioned ordinal as a selector.